editor/src/components/common/shared-strategies/convert-to-grid-strategy.ts

```diff
@@ -0,0 +1,118 @@
+import type { ElementPath } from 'utopia-shared/src/types'
+import type { ElementPathTrees } from '../../../core/shared/element-path-tree'
+import type { ElementInstanceMetadataMap } from '../../../core/shared/element-template'
+import type { AllElementProps } from '../../editor/store/editor-state'
+import type { CanvasCommand } from '../../canvas/commands/commands'
+import {
+  flexContainerProps,
+  gridContainerProps,
+  nukeAllAbsolutePositioningPropsCommands,
+  prunePropsCommands,
+  sizeToVisualDimensions,
+} from '../../inspector/inspector-common'
+import { getChildrenPathsForContainer } from './convert-strategies-common'
+import type { CanvasFrameAndTarget } from '../../canvas/canvas-types'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { setProperty } from '../../canvas/commands/set-property-command'
+import * as PP from '../../../core/shared/property-path'
+
+function guessLayoutInfoAlongAxis(
+  children: Array<CanvasFrameAndTarget>,
+  sortFn: (a: CanvasFrameAndTarget, b: CanvasFrameAndTarget) => number,
+  comesAfter: (a: CanvasFrameAndTarget, b: CanvasFrameAndTarget) => boolean,
+  gapBetween: (a: CanvasFrameAndTarget, b: CanvasFrameAndTarget) => number,
+): { nChildren: number; averageGap: number } {
+  if (children.length === 0) {
+    return { nChildren: 0, averageGap: 0 }
+  }
+
+  const sortedChildren = children.sort(sortFn)
+  let childrenAlongAxis = 1
+  let gaps: number[] = []
+  let currentChild = sortedChildren[0]
+  for (const child of sortedChildren.slice(1)) {
+    if (comesAfter(currentChild, child)) {
+      childrenAlongAxis += 1
+      gaps.push(gapBetween(currentChild, child))
+      currentChild = child
+    }
+  }
+
+  const averageGap =
+    gaps.length === 0 ? 0 : Math.floor(gaps.reduce((a, b) => a + b, 0) / gaps.length)
+
+  return {
+    nChildren: childrenAlongAxis,
+    averageGap: averageGap,
+  }
+}
+
+function guessMatchingGridSetup(children: Array<CanvasFrameAndTarget>): {
+  gap: number
+  numberOfColumns: number
+  numberOfRows: number
+} {
+  const horizontalData = guessLayoutInfoAlongAxis(
+    children,
+    (a, b) => a.frame.x - b.frame.x,
```

Comment:
This works, and I guess is fine, but it might have been clearer to pass in an object with named keys for the various functions. Not worth changing in the PR now, just something to keep in mind for the future.